File: backend/src/main/java/com/soufiane/open_data/dtos/dataSetTheme/DataSetThemeRequest.java
package com.soufiane.open_data.dtos.dataSetTheme;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor
public class DataSetThemeRequest {
    private String name;
    private String description;
    private String createdBy;
    private String updatedBy;
    private LocalDateTime createdOn;
    private LocalDateTime updatedOn;
    private byte[] iconData;
    private String iconPath;
    private String icon;
}

File: frontend/src/app/models/DataSetThemeRequest.ts
export interface DataSetThemeRequest {
  uuid: string; 
  name: string;
  description: string;
  createdBy: string;
  updatedBy: string;
  createdOn: Date;
  updatedOn: Date;
  iconData?: string;   
  iconPath?: string;   
  icon: string;
}